### PR TITLE
[Merged by Bors] - feat(Condensed): light condensed modules satisfy countable AB4*

### PR DESCRIPTION
**Pull request by dagurtomas**
Given sequences `M N : ℕ → C` of objects with morphisms `f n : M n ⟶ N n` for all `n`, we exhibit `∏ M` as the limit of the tower

```
⋯ → ∏_{n < m + 1} M n × ∏_{n ≥ m + 1} N n → ∏_{n < m} M n × ∏_{n ≥ m} N n → ⋯ → ∏ N
```
Further, the transition maps in this tower are epimorphisms, in the case when each `f n` is an epimorphism and `C` has finite biproducts.

We use this to conclude that light condensed modules over a ring R satisfy a countable version of Grothendieck's AB4* axiom.

---

- [x] depends on: #18336
- [x] depends on: #19190 
- [x] depends on: #19195  
- [x] depends on: #19199 
- [x] depends on: #19200 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Limits/Shapes/SequentialProduct.lean [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+  rw [functorMap, Pi.map_eq_prod_map (P := fun m : ℕ ↦ m < n + 1)]
+  apply (config := { allowSynthFailures := true }) epi_comp
+  apply (config := { allowSynthFailures := true }) epi_comp
+  apply (config := { allowSynthFailures := true }) prod.map_epi
+  · apply (config := { allowSynthFailures := true }) Pi.map_epi
+    intro ⟨_, _⟩
+    split
+    all_goals infer_instance
+  · apply (config := { allowSynthFailures := true }) IsIso.epi_of_iso
+    apply (config := { allowSynthFailures := true }) Pi.map_isIso
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Limits/Shapes/SequentialProduct.lean:224]:
```suggestion
  apply +allowSynthFailures epi_comp
  apply +allowSynthFailures epi_comp
  apply +allowSynthFailures prod.map_epi
  · apply +allowSynthFailures Pi.map_epi
    intro ⟨_, _⟩
    split
    all_goals infer_instance
  · apply +allowSynthFailures IsIso.epi_of_iso
    apply +allowSynthFailures Pi.map_isIso
```

**Review by jcommelin — COMMENTED** [review]
LGTM

bors d+

**Comment by jcommelin** [comment]
Hmm, I'm confused why the build is failing. I thought this syntax would work. Could you please merge master?

**Comment by dagurtomas** [comment]
Merging didn't work, do I need some import for this syntax?

**Comment by dagurtomas** [comment]
Let's just use the old syntax

bors merge

**Pull request closed without merging**